web/src/user_groups.ts

```diff
@@ -332,6 +332,12 @@ export function is_subgroup_of_target_group(target_group_id: number, subgroup_id
     return recursive_subgroup_ids.has(subgroup_id);
 }
 
+export function get_supergroups_of_user_group(group_id: number): UserGroup[] {
+    return get_realm_user_groups().filter((user_group) =>
+        is_subgroup_of_target_group(user_group.id, group_id),
+    );
+}
```

Comment:
This is not the best algorithm; ideally we'd do a floodfill like `get_recursive_group_members` that doesn't process any edge more than once. Can fix in a follow-up; that sort of non-optimal approach is pervasive in this file.